controller/src/test/java/io/pravega/controller/server/rpc/auth/ControllerGrpcAuthFocusedTest.java

```diff
@@ -573,4 +614,50 @@ private static File createAuthFile() {
         private final static String SCOPE1_STREAM2_READ = "authSc1Str2readonly";
         private final static String SCOPE1_STREAM1_LIST_READ = "scope1stream1lr";
     }
+
+    static class ResultObserver<T> implements StreamObserver<T> {
+        private T result = null;
+        private Throwable error;
+        private final AtomicBoolean completed = new AtomicBoolean(false);
+
+        @Override
+        public void onNext(T value) {
+            result = value;
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            synchronized (this) {
+                error = t;
+                completed.set(true);
+                this.notifyAll();
+            }
+        }
+
+        @Override
+        public void onCompleted() {
+            synchronized (this) {
+                completed.set(true);
+                this.notifyAll();
+            }
+        }
+
+        @SneakyThrows
+        public T get() {
+            synchronized (this) {
+                while (!completed.get()) {
+                    try {
+                        this.wait();
```

Comment:
It's bad practice to block while holding a lock.

Similarly above, you invoke `notifyAll` while holding the lock.

Reply:
The modified code doesn't use these.